common/bbs/search.c

```diff
@@ -38,10 +38,13 @@ match_fileheader_predicate(const fileheader_t *fh, void *arg)
 	return DBCS_strcasestr(fh->title, keyword) == NULL;
     else if (sr_mode & RS_TITLE)
 	return strcasecmp(subject(fh->title), keyword) == 0;
-    else if (sr_mode & RS_RECOMMEND)
-	return pred->recommend > 0 ?
+    else if (sr_mode & RS_RECOMMEND) {
+        if (fh->filemode & FILE_SUSPICIOUS)
+            return 0;
```

Comment:
Can this entry be treated as `fh->recommend` being `0`, not silently pretending it is not matching?

Reply:
Thanks. I changed the implementation to treat the file as ```fh->recommend``` being ```0```.